### DISCUSSION TITLE
Hydration mismatch error with local-only data component, how to solve?

### DISCUSSION THREAD

**Discussion by killerchip** [vercel/next.js#45954] [Help] [answered]
Hello.

I'm relatively new to NextJS and I'm trying to understand how to overcome the following issue.

I have a page which displays a random image. The image URL is created on server side with `getServerSideProps`.
The page apart from displaying the image, it also 'pushes' it to a `mobx` store which keeps the user's history of images in 'localStorage'.

The same page includes a 'History' component which displays the user's image browse history. This component obtains the data from the mobx store, which when constructed it loads the URLs from the LocalStorage.

Of course the first render works fine. But when I try to 'deeplink' again to this page, I get the 
`Error: Hydration failed because the initial UI does not match what was rendered on the server.`
because the 'fetched' page has an Empty History component (server can't know the user's history), while local render includes a  'History' component that is displaying data from the local storage.

How I can overcome this limitation/error? Am I missing something on how this should be architectured?

Thanks in advance

**Comment by guitheengineer** [comment, marked as answer]
In this case if there's nothing you can do about it, suppress the error with suppressHydrationWarning or render it only on the client

```jsx
function ClientOnly({ children, ...props }) {
  const [isMounted, setIsMounted] = React.useState(false);
  React.useEffect(() => {
    setIsMounted(true);
  }, []);
  if (!isMounted) {
    return null;
  }
  return (
    <div {...props}>
      {children}
    </div>
  );
}
```

**Reply by killerchip** [reply]
Thank you very much for your answer.

Render only on client after mounting, sounds like a good general plan for such cases !
